cluster-autoscaler/cloudprovider/test/test_cloud_provider.go

```diff
@@ -250,10 +308,24 @@ func (tng *TestNodeGroup) Nodes() ([]string, error) {
 
 // Autoprovisioned returns true if the node group is autoprovisioned.
 func (tng *TestNodeGroup) Autoprovisioned() bool {
-	return false
+	return tng.autoprovisioned
 }
 
 // TemplateNodeInfo returns a node template for this node group.
 func (tng *TestNodeGroup) TemplateNodeInfo() (*schedulercache.NodeInfo, error) {
-	return nil, cloudprovider.ErrNotImplemented
+	if tng.cloudProvider.machineTemplates == nil {
+		return nil, cloudprovider.ErrNotImplemented
```

Comment:
Does ErrNotImplemented indicate there's a plan to handle it differently in the future?

Reply:
no, it mimics node group that don't support templatenodeinfo and ErrNotImplemented is the most probable error returned from them.